### DISCUSSION TITLE
NES/Famicom option to dump headerless (raw)

### DISCUSSION THREAD

**Discussion by VincentEmond** [sanni/cartreader#451] [Ideas & Suggestions]
Hello,

I noticed that on the wiki page it says:

> The Cart Reader will output both a *.bin and *.nes file. The *.bin file contains just the maskrom data of the cartridge while the *.nes file has an addition iNES emulation header added. This header is not found in the cartridge. The Cart Reader has a database with all the emulation headers and tries to add the header to the bin file after dumping but it might not recognize all the games.

However running `HW5 FW:9.1`  this is not the behavior I get from the reader. I only get a `*.nes` file but no `*.bin` file. I know the dump is already checked for validity against the database on the SDCard but I always like calculating and verifying the checksums myself against nointro. In that case nointro seem to only keep track of the headerless (raw) dump or at least the headered version (.nes) I get doesn't match anything in their DB.

I have modified the code (copy pasted to say the truth ;) ) to add a `Read Raw` option. This options reads the `PRG` and `CHR` chips and merge them together to get a `.raw` file.

I guess I could always use the `Dump PRG` and `Dump CHR` options but I am lazy and I would rather let the reader merge these sections for me. It's not perfect either because some games have a `CHR RAM` instead of `CHR ROM` so in that case then guess you just want to keep the `PRG` content instead of both.

Anyways, I just thought I'd share the code if anybody is interested
[NES.zip](https://github.com/sanni/cartreader/files/9239084/NES.zip)
.

**Comment by sanni** [comment]
I added the readRaw_NES() function to the code, but I put it in the Read PRG/CHR submenu. 👍

**Reply by VincentEmond** [reply]
The PRG/CHR menu is a better place to put it indeed, I should have thought about that! I guess in that case it could be renamed to `Read PRG+CHR` to be more specific. I also removed the call to the CRC check function from that option because I wasn't sure if it was going to match anything. If the database contains the headerless checksums then the call to this function should be included back.
